### PR TITLE
Bump Core-Interface in client packages

### PR DESCRIPTION
**Pull request by anthony-murphy**
related to #8076

### CODE REVIEW

File: api-report/file-driver.api.md [hunk 1]
```diff
@@ -90,6 +90,8 @@ export const FileSnapshotWriterClassFactory: <TBase extends ReaderConstructor>(B
         createBlob(file: ArrayBufferLike): Promise<api.ICreateBlobResponse>;
         uploadSummaryWithContext(summary: api.ISummaryTree, context: ISummaryContext): Promise<string>;
         downloadSummary(handle: api.ISummaryHandle): Promise<api.ISummaryTree>;
+        readonly disposed?: boolean | undefined;
```
Comment (anthony-murphy) [inline api-report/file-driver.api.md:93]: this is due to @znewton change coming in

File: packages/loader/container-loader/src/deltaManager.ts [hunk 2]
```diff
@@ -1322,14 +1322,7 @@ export class DeltaManager
...
 
-        // back-compat: added in 0.45. Make it unconditional (i.e. use connection.dispose()) in some future.
-        const disposable = connection as Partial<IDisposable>;
-
-        if (disposable.dispose !== undefined) {
-            disposable.dispose();
-        } else {
-            connection.close();
-        }
+        connection.dispose();
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/deltaManager.ts:1325]: need this due to @scottn12 change #8185

Comment (anthony-murphy) [reply packages/loader/container-loader/src/deltaManager.ts:1325]: @heliocliu this is a good example of a problem with our unlinked repo

**Review by DLehenbauer — APPROVED** [review]
pre-approved

**Review by andre4i — APPROVED** [review]

**Pull request merged**